From 04d8355ac606cc96390a51d3db1152aafd4747c6 Mon Sep 17 00:00:00 2001
From: Fangrui Song <maskray@google.com>
Date: Thu, 4 Nov 2021 00:55:21 -0700
Subject: [PATCH 67/97] readelf: Make DT_PREINIT_ARRAYSZ's output style match
 DT_INIT_ARRAYSZ

The output now looks like:

- 0x0000000000000021 (PREINIT_ARRAYSZ)    0x10
+ 0x0000000000000021 (PREINIT_ARRAYSZ)    16 (bytes)
  0x0000000000000019 (INIT_ARRAY)         0xbefc90
  0x000000000000001b (INIT_ARRAYSZ)       536 (bytes)

	* readelf.c (process_dynamic_section): Handle DT_PREINIT_ARRAYSZ.
---
 binutils/readelf.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/binutils/readelf.c b/binutils/readelf.c
index 0fed82b63e3..7a162eead17 100644
--- a/binutils/readelf.c
+++ b/binutils/readelf.c
@@ -11385,6 +11385,7 @@ the .dynstr section doesn't match the DT_STRTAB and DT_STRSZ tags\n"));
 	case DT_PLTPADSZ:
 	case DT_MOVEENT	:
 	case DT_MOVESZ	:
+	case DT_PREINIT_ARRAYSZ:
 	case DT_INIT_ARRAYSZ:
 	case DT_FINI_ARRAYSZ:
 	case DT_GNU_CONFLICTSZ:
-- 
2.34.0

